src/core/Akka/Dispatch/AbstractDispatcher.cs

```diff
@@ -293,7 +374,39 @@ private bool UpdateShutdownSchedule(int expected, int update)
         }
 
         /// <summary>
-        ///     Schedules the specified run.
+        /// Used for executing arbitrary <see cref="IRunnable"/>s on this <see cref="MessageDispatcher"/>.
+        /// 
+        /// Not used for mailbox runs.
+        /// </summary>
+        class UnbatchedExecute : IRunnable
```

Comment:
A special type of `IRunnable` used for executing arbitrary tasks on the dispatcher. **Not used for executing the mailbox`. If we were to expose a custom `TaskScheduler` that ran on a `MessageDispatcher`, it would use this method as it would guarantee that the dispatcher couldn't accidentally shut down before all scheduled tasks were complete.